velox/functions/lib/string/StringCore.h

```diff
@@ -194,6 +195,18 @@ FOLLY_ALWAYS_INLINE size_t lowerUnicode(
     }
 
     inputIdx += size;
+
+    if constexpr (forSpark) {
+      // Handle Turkish-specific case for İ (U+0130).
+      if (UNLIKELY(nextCodePoint == 0x0130)) {
```

Comment:
Do you try other combination dot above?https://www.compart.com/en/unicode/U+0307 ,such as T with dot above.